tatanka/tatanka.go

```diff
@@ -183,22 +194,36 @@ func New(cfg *Config) (*Tatanka, error) {
 		}
 	}
 
+	const maxFiatRateBroadcastIntervalInMinutes = 8
+	if cfg.FiatRateBroadcastInterval > maxFiatRateBroadcastIntervalInMinutes || cfg.FiatRateBroadcastInterval == 0 {
+		cfg.FiatRateBroadcastInterval = maxFiatRateBroadcastIntervalInMinutes
+	}
+
 	t := &Tatanka{
-		net:           cfg.Net,
-		dataDir:       cfg.DataDir,
-		log:           cfg.Logger,
-		whitelist:     whitelist,
-		db:            db,
-		priv:          priv,
-		id:            peerID,
-		chains:        chains,
-		tatankas:      make(map[tanka.PeerID]*remoteTatanka),
-		clients:       make(map[tanka.PeerID]*client),
-		remoteClients: make(map[tanka.PeerID]map[tanka.PeerID]struct{}),
-		topics:        make(map[tanka.Topic]*Topic),
-		recentRelays:  make(map[[32]byte]time.Time),
-		clientJobs:    make(chan *clientJob, 128),
+		net:                       cfg.Net,
+		dataDir:                   cfg.DataDir,
+		log:                       cfg.Logger,
+		whitelist:                 whitelist,
+		db:                        db,
+		priv:                      priv,
+		id:                        peerID,
+		chains:                    chains,
+		tatankas:                  make(map[tanka.PeerID]*remoteTatanka),
+		clients:                   make(map[tanka.PeerID]*client),
+		remoteClients:             make(map[tanka.PeerID]map[tanka.PeerID]struct{}),
+		topics:                    make(map[tanka.Topic]*Topic),
+		recentRelays:              make(map[[32]byte]time.Time),
+		clientJobs:                make(chan *clientJob, 128),
+		fiatRateBroadcastInterval: cfg.FiatOracleConfig.FiatRateBroadcastInterval,
```

Comment:
Right now you have rate oracle syncing in background on it's own refresh schedule and tatanka periodically grabbing the rates (of varying degrees of staleness) and broadcasts them. I'd rather tatanka receive rates via channel immediate when they are updated by the oracle and distribute them immediately.